src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat.slice(3);
+  const newSeparator = toFormat.slice(3);
+  const splitDate = date.split(oldSeparator);
+
+  const newDate = [];
+  const objectDate = {};
+
+  for (let i = 0; i < splitDate.length; i++) {
+    objectDate[fromFormat[i]] = splitDate[i];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (objectDate['YY'] < 30) {
+      objectDate['YYYY'] = '20' + objectDate['YY'];
+    } else {
+      objectDate['YYYY'] = '19' + objectDate['YY'];
+    }
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    objectDate['YY'] = objectDate['YYYY'].slice(-2);
+  }
+
+  for (let i = 0; i < splitDate.length; i++) {
+    newDate[i] = objectDate[toFormat[i]];
+  }
+
+  return newDate.join(newSeparator);
```

Comment:
The `join` method is called with `newSeparator`, which is currently an array due to the incorrect use of `slice` on line 12. After fixing the `slice` call, make sure to pass the actual string separator to the `join` method.